daml/ContingentClaims/Claim.daml

```diff
@@ -72,109 +71,90 @@ mapParams ft' ft fk fv =
+  | AndF with claims: [b]
+  | OrF with claims: [b]
+  | CondF with predicate: Inequality t x a, success: b, failure: b
+  | ScaleF with k: Observation t x a, claim: b
+  | WhenF with predicate: Inequality t x a, claim: b
+  | AnytimeF with predicate: Inequality t x a, claim: b
+  | UntilF with predicate: Inequality t x a, claim: b
   deriving (Functor)
 
-instance Recursive (Claim f t x a) (ClaimF f t x a) where
+instance Recursive (Claim t x a) (ClaimF t x a) where
   project Zero = ZeroF
   project (One a) = OneF a
   project (Give c) = GiveF c
-  project (And c c') = (AndF c c')
-  project (Or c c') = (OrF c c')
+  project (And cs) = (AndF cs)
+  project (Or cs) = (OrF cs)
   project (Cond p c c') = CondF p c c'
   project (Scale k c) = ScaleF k c
   project (When p c) = WhenF p c
   project (Anytime p c) = AnytimeF p c
   project (Until p c) = UntilF p c
 
-instance Corecursive (Claim f t x a) (ClaimF f t x a) where
+instance Corecursive (Claim t x a) (ClaimF t x a) where
   embed ZeroF = Zero
   embed (OneF x) = One x
   embed (GiveF a) = Give a
-  embed (AndF c c') = (And c c')
-  embed (OrF c c') = (Or c c')
+  embed (AndF cs) = (And cs)
+  embed (OrF cs) = (Or cs)
   embed (CondF p c c') = Cond p c c'
   embed (ScaleF k c) = Scale k c
   embed (WhenF p c) = When p c
   embed (AnytimeF p c) = Anytime p c
   embed (UntilF p c) = Until p c
 
-instance Foldable (ClaimF f t x a) where
+instance Foldable (ClaimF t x a) where
   foldr _ seed ZeroF = seed
   foldr _ seed (OneF _) = seed
   foldr f seed (GiveF c) = f c seed
   foldr f seed (WhenF _ c) = f c seed
   foldr f seed (ScaleF _ c) = f c seed
-  foldr f seed (AndF c c') = f c $ f c' seed
-  foldr f seed (OrF c c') = f c $ f c' seed
+  foldr f seed (AndF cs) = Prelude.foldr f seed cs
+  foldr f seed (OrF cs) = Prelude.foldr f seed cs
   foldr f seed (CondF _ c c') = f c $ f c' seed
   foldr f seed (AnytimeF _ c) = f c seed
   foldr f seed (UntilF _ c) = f c seed
 
-instance Traversable (ClaimF g t x b) where
+instance Traversable (ClaimF t x b) where
   sequence ZeroF = pure ZeroF
   sequence (OneF asset) = pure $ OneF asset
   sequence (GiveF fa) = GiveF <$> fa
   sequence (WhenF p fa) = WhenF p <$> fa
   sequence (ScaleF p fa) = ScaleF p <$> fa
-  sequence (AndF fa fa') = AndF <$> fa <*> fa'
-  sequence (OrF fa fa') = OrF <$> fa <*> fa'
+  sequence (AndF [fa]) = (\a -> AndF [a]) <$> fa
+  sequence (AndF (fa :: fas)) =  and <$> fa <*> Prelude.sequence fas
+    where and a as = AndF (a :: as)
+  sequence (AndF []) = error "Traversable ClaimF : sequence empty AndF"
+  sequence (OrF [fa]) = (\a -> OrF [a]) <$> fa
+  sequence (OrF (fa :: fas)) = or <$> fa <*> Prelude.sequence fas
+    where or a as = OrF (a :: as)
+  sequence (OrF []) = error "Traversable ClaimF : sequence empty OrF"
   sequence (CondF p fa fa') = CondF p <$> fa <*> fa'
   sequence (AnytimeF p fa) = AnytimeF p <$> fa
   sequence (UntilF p fa) = UntilF p <$> fa
+
+-- Inequality --
```

Comment:
Should I put this in it's own file?

Reply:
This would probably make a nicer structure, but the fact is that `Inequality` is only used in the creation of a `Claim`, so this would just be one more thing people would need to know. In case of a `Claim` blanket import, you don't need to even be aware of this.